src/txn/txn_timestamp.c

```diff
@@ -463,6 +467,23 @@ __wt_txn_global_set_timestamp(WT_SESSION_IMPL *session, const char *cfg[])
     WT_RET(__wt_txn_parse_timestamp(session, "oldest", &oldest_ts, &oldest_cval));
     WT_RET(__wt_txn_parse_timestamp(session, "stable", &stable_ts, &stable_cval));
 
+    /*
+     * We set the oldest timestamp to the greater of the supplied oldest timestamp and the oldest
+     * timestamp in the history store checkpoint list. Note that the oldest history store timestamp
+     * is set during recovery and that if set, we reset it to WT_TS_NONE to ensure this code is only
+     * executed on a restart. If the checkpoint oldest timestamp is used, ensure the stable
+     * timestamp is at least as large.
+     */
+    if (txn_global->oldest_ckpt_hs_timestamp > oldest_ts) {
```

Comment:
@agorrod, should we just return error if the user supplied oldest timestamp is smaller than the oldest checkpoint timestamp or should we update the oldest timestamp ?

Reply:
I think we should return an EINVAL error.

Reply:
Thanks @agorrod, I'll update the code.

Reply:
Thanks! I'm going to take myself off the balance of this review.